src/window/basic_frame.rs

```diff
@@ -228,7 +229,7 @@ impl Frame for BasicFrame {
         compositor: &Proxy<wl_compositor::WlCompositor>,
         subcompositor: &Proxy<wl_subcompositor::WlSubcompositor>,
         shm: &Proxy<wl_shm::WlShm>,
-        implementation: Box<Implementation<u32, FrameRequest> + Send>,
+        implementation: Box<FnMut(u32, FrameRequest) + Send>,
```

Comment:
I'd prefer the arguments of this closure swapped to. Main matter (the requests) first, additional data second.